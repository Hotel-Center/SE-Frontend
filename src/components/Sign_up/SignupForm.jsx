// import * as React from "react";
import Button from "@mui/material/Button";
import TextField from "@mui/material/TextField";
import Box from "@mui/material/Box";
import Link from "@mui/material/Link";
import "./style.css";
import { Sign_up_connection } from "../../Utils/connection";
import { useFormik } from "formik";
import Stack from "@mui/material/Stack";

function SignupForm(props) {
  const formik = useFormik({
    initialValues: {
      email: "",
      password: "",
      password_confirm: "",
    },
    validate: function (values) {
      const errors = {};
      if (!values.email) {
        errors.email = "*Email field must not be empty!";
      } else if (
        !values.email.match(
          /^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i
        )
      ) {
        errors.email = "*Please enter a valid email.";
      }

      if (!values.password) {
        errors.password = "*Password field must not be empty";
      } else if (values.password.length < 8) {
        errors.password = "*Password length should be at least 8 characters.";
      } else if (!values.password.match(/^.*(?=.*[@#$%&]).*$/)) {
        errors.password =
          "*Password should contain at least one special character ( @, #, %, &,  $,).";
      } else if (!values.password.match(/^.*(?=.*[A-Z]).*$/)) {
        errors.password =
          "*Password should contain at least one uppercase letter(A-Z).";
      } else if (!values.password.match(/^.*(?=.*[a-z]).*$/)) {
        errors.password =
          "*Password should contain at least one lowercase letter(a-z).";
      } else if (!values.password.match(/^.*(?=.*\d).*$/)) {
        errors.password = "*Password should contain at least one digit(0-9). ";
      }

      if (!values.password_confirm) {
        errors.password_confirm = "*This field must not be empty!";
      } else if (values.password_confirm != values.password) {
        errors.password_confirm = "*Confirm password doesnt match!";
      }

      return errors;
    },
    onSubmit: function (values) {
      props.dispatch({
        type: "set_login_info",
        value: values,
      });
      const is_registered = Sign_up_connection(values.email, values.password);

      if (is_registered) {
        window.location.replace("/verify-email");
      }
    },
  });

  return (
    <Box
      fullWidth
      component="form"
      noValidate
      onSubmit={formik.handleSubmit}
      sx={{ height: "100%", display: "flex", flexDirection: "column" }}
    >
      <Stack spacing={1} fullWidth>
        <TextField
          required
          fullWidth
          id="email"
          label="Email Address"
          name="email"
          autoComplete="email"
          className={
            formik.errors.email ? "is-invalid form-control" : "form-control"
          }
          value={formik.values.email}
          onChange={formik.handleChange}
        />
        <p className="err">
          {formik.errors.email || " "}
          <br />
        </p>
        <TextField
          required
          fullWidth
          name="password"
          label="Password"
          type="password"
          id="password"
          autoComplete="new-password"
          className={
            formik.errors.password ? "is-invalid form-control" : "form-control"
          }
          value={formik.values.password}
          onChange={formik.handleChange}
        />
        <p className="err">
          {formik.errors.password || " "}
          <br />
        </p>
        <TextField
          required
          fullWidth
          name="password_confirm"
          label="Confirm Password"
          type="password"
          id="password_confirm"
          autoComplete="new-password"
          className={
            formik.errors.password_confirm
              ? "is-invalid form-control"
              : "form-control"
          }
          value={formik.values.password_confirm}
          onChange={formik.handleChange}
        />
        <p className="err">
          {formik.errors.password_confirm || " "}
          <br />
        </p>
      </Stack>
      <Box
        fullWidth
        sx={{
          display: "flex",
          alignItems: "center",
          gap: 3,
          mt: "auto",
          mb: 2,
        }}
      >
        <Button
          sx={{
            height: "40px",
            width: "45%",
          }}
          variant="outlined"
          onClick={props.prevStep}
        >
          Back
        </Button>
        <Button
          sx={{
            height: "40px",
            width: "45%",
            ml: "auto",
          }}
          type="submit"
          variant="contained"
        >
          Submit
        </Button>
      </Box>
      <Link
        to="./login"
        variant="body2"
        sx={{
          color: "black",
          mt: 2,
        }}
      >
        Already have an account? Sign in
      </Link>
    </Box>
  );
}

export default SignupForm;
