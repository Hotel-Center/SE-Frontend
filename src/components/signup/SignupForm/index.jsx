// import * as React from "react";
import Button from "@mui/material/Button";
import TextField from "@mui/material/TextField";
import Box from "@mui/material/Box";
import Link from "@mui/material/Link";
import styles from "../style.module.css";
import { Sign_up_connection } from "../../../Utils/connection";
import { useFormik } from "formik";
import Stack from "@mui/material/Stack";

function SignupForm(props) {
  const formik = useFormik({
    initialValues: {
      email: "",
      phone_number: "",
      password: "",
      password_confirm: "",
    },
    validate: function (values) {
      const errors = {};
      if (!values.email) {
        errors.email = "*Email field must not be empty!";
      } else if (
        !values.email.match(
          /^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i
        )
      ) {
        errors.email = "*Please enter a valid email.";
      }

      if (!values.phone_number) {
        errors.phone_number = "*Phone number must not be empty!";
      }

      if (!values.password) {
        errors.password = "*Password field must not be empty";
      } else if (values.password.length < 8) {
        errors.password = "*Password length should be at least 8 characters.";
      } else if (!values.password.match(/^.*(?=.*[@#$%&]).*$/)) {
        errors.password =
          "*Password should contain at least one special character ( @, #, %, &,  $,).";
      } else if (!values.password.match(/^.*(?=.*[A-Z]).*$/)) {
        errors.password =
          "*Password should contain at least one uppercase letter(A-Z).";
      } else if (!values.password.match(/^.*(?=.*[a-z]).*$/)) {
        errors.password =
          "*Password should contain at least one lowercase letter(a-z).";
      } else if (!values.password.match(/^.*(?=.*\d).*$/)) {
        errors.password = "*Password should contain at least one digit(0-9). ";
      }

      if (!values.password_confirm) {
        errors.password_confirm = "*This field must not be empty!";
      } else if (values.password_confirm != values.password) {
        errors.password_confirm = "*Confirm password doesnt match!";
      }

      return errors;
    },
    onSubmit: async function (values) {
      props.dispatch({
        type: "set_login_info",
        value: values,
      });
      const is_registered = await Sign_up_connection(
        values.email,
        values.phone_number,
        props.role,
        values.password
      );

      if (is_registered) {
        window.location.replace("/verify-email");
      }
    },
  });

  return (
    <Box
      component="form"
      noValidate
      onSubmit={formik.handleSubmit}
      sx={{ height: "100%", display: "flex", flexDirection: "column" }}
    >
      <Stack spacing={1}>
        <TextField
          required
          fullWidth
          id="email"
          label="Email Address"
          name="email"
          autoComplete="email"
          className={
            formik.errors.email ? "is-invalid form-control" : "form-control"
          }
          value={formik.values.email}
          onChange={formik.handleChange}
        />
        <p className={styles.err}>
          {formik.errors.email || " "}
          <br />
        </p>
        <TextField
          required
          fullWidth
          id="phone_number"
          label="Phone Number"
          name="phone_number"
          autoComplete="phone"
          className={
            formik.errors.phone_number
              ? "is-invalid form-control"
              : "form-control"
          }
          value={formik.values.phone_number}
          onChange={formik.handleChange}
        />
        <p className={styles.err}>
          {formik.errors.phone_number || " "}
          <br />
        </p>
        <TextField
          required
          fullWidth
          name="password"
          label="Password"
          type="password"
          id="password"
          autoComplete="new-password"
          className={
            formik.errors.password ? "is-invalid form-control" : "form-control"
          }
          value={formik.values.password}
          onChange={formik.handleChange}
        />
        <p className={styles.err}>
          {formik.errors.password || " "}
          <br />
        </p>
        <TextField
          required
          fullWidth
          name="password_confirm"
          label="Confirm Password"
          type="password"
          id="password_confirm"
          autoComplete="new-password"
          className={
            formik.errors.password_confirm
              ? "is-invalid form-control"
              : "form-control"
          }
          value={formik.values.password_confirm}
          onChange={formik.handleChange}
        />
        <p className={styles.err}>
          {formik.errors.password_confirm || " "}
          <br />
        </p>
      </Stack>
      <Link
        href="/login"
        variant="body2"
        sx={{
          color: "black",
          mt: "auto",
        }}
      >
        Already have an account? Sign in
      </Link>
      <Box
        sx={{
          display: "flex",
          alignItems: "center",
          gap: 3,
          mt: 2,
          mb: 2,
        }}
      >
        <Button
          sx={{
            height: "40px",
            width: "45%",
          }}
          variant="outlined"
          onClick={props.prevStep}
        >
          Back
        </Button>
        <Button
          sx={{
            height: "40px",
            width: "45%",
            ml: "auto",
          }}
          type="submit"
          variant="contained"
        >
          Signup
        </Button>
      </Box>
    </Box>
  );
}

export default SignupForm;
