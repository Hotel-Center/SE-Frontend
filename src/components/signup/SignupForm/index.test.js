import renderer from "react-test-renderer";
import SignupForm from ".";

describe("Signup Form", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SignupForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
