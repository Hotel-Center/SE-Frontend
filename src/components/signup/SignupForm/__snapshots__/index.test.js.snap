// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Form renders correctly 1`] = `
<form
  className="MuiBox-root css-1sm2s1z"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiStack-root css-1okj3ks-MuiStack-root"
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root form-control css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="email"
        id="email-label"
      >
        Email Address
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="email"
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="email"
          name="email"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Email Address
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <p
      className="err"
    >
       
      <br />
    </p>
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root form-control css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="phone_number"
        id="phone_number-label"
      >
        Phone Number
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="phone"
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="phone_number"
          name="phone_number"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Phone Number
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <p
      className="err"
    >
       
      <br />
    </p>
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root form-control css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="password"
        id="password-label"
      >
        Password
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="new-password"
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="password"
          name="password"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Password
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <p
      className="err"
    >
       
      <br />
    </p>
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root form-control css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="password_confirm"
        id="password_confirm-label"
      >
        Confirm Password
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="new-password"
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="password_confirm"
          name="password_confirm"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Confirm Password
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <p
      className="err"
    >
       
      <br />
    </p>
  </div>
  <a
    className="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways css-eb54fy-MuiTypography-root-MuiLink-root"
    href="/login"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    Already have an account? Sign in
  </a>
  <div
    className="MuiBox-root css-1v6o1we"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1sj7b83-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Back
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-kq49rb-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      Signup
    </button>
  </div>
</form>
`;
